smarts/core/sumo_traffic_simulation.py

```diff
@@ -331,6 +354,17 @@ def teardown(self):
         self._to_be_teleported = dict()
         self._reserved_areas = dict()
 
+    @property
+    def connected(self):
+        return self._traci_conn != None
+
+    @property
+    def recovery_options(self) -> ProviderRecoveryOptions:
+        return (
+            ProviderRecoveryOptions.EPISODE_REQUIRED
+            | ProviderRecoveryOptions.ATTEMPT_RECOVERY
+        )
```

Comment:
This means if `SUMO TraCI` connection fails, we need to rerun that episode or try to attempt to reconnect to `TraCI`?`

Reply:
It will attempt  `SumoTrafficSimulation.recover()` for now which just currently disconnects or re-throws the exception from the provider.